### PR TITLE
Add docbook-tool JS

### PR DESCRIPTION
**Pull request by cspray**

### CODE REVIEW

File: src/docbook-tool.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            selectedListElement.classList.remove('hidden');
+        }
+    }
+
+    function fragmentRoute(pathWithHash) {
+        const path = pathWithHash.substring(1);
+
+        const pathParts = path.split("/");
+
+        let pageTitle = 'WISP Reporting Tool Docs';
```
Comment (asgrim) [inline src/docbook-tool.js:34]: This is the page title I mentioned that needs to change depending on what is using it :thinking:

File: src/docbook-tool.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+"use strict";
```
Comment (asgrim) [inline src/docbook-tool.js:1]: Can we add a comment what this is for please? Ideally something to stop someone coming along later down the line who forgot what this thing is for and deleting it, thus breaking downstream systems

Comment (cspray) [reply src/docbook-tool.js:1]: _starts looking up why we put this in here_

File: src/docbook-tool.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            selectedListElement.classList.remove('hidden');
+        }
+    }
+
+    function fragmentRoute(pathWithHash) {
+        const path = pathWithHash.substring(1);
+
+        const pathParts = path.split("/");
+
+        let pageTitle = document.title;
```
Comment (asgrim) [inline src/docbook-tool.js:34]: Oh - I think this would result in an endlessly growing page title :/ it needs to be staticly defined :thinking:

**Review by Ocramius — CHANGES_REQUESTED** [review]
What about hosting the DocbookTool JS directly in the `gh-pages` branch of DocbookTool?

File: src/docbook-tool.js (hunk 3 above)
Comment (cspray) [reply src/docbook-tool.js:34]: Oh, yea you're probably right.... hrmm.... 🤔

File: src/docbook-tool.js (hunk 2 above)
Comment (asgrim) [reply src/docbook-tool.js:1]: Not specifically `use strict` - I mean the whole file in general :grin:

Comment (cspray) [reply src/docbook-tool.js:1]: Oh, lol. That makes more sense.

**Comment by cspray** [comment]
Decided we're gonna put this into DocBookTool directly and not rely on roave.com for hosting this.

**Pull request closed without merging**